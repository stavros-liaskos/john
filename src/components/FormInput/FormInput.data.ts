export const formInputI18n = {
  label: 'Here goes the artist..',
};
