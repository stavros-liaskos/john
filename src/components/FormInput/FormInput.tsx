import React, { useState } from 'react';

type FormInputProps = {
  handleAction: (input: string) => void;
  i18n: {
    label: string;
  };
  actionEventTrigger: 'onChange' | 'onSubmit';
  children?: React.ReactNode;
};

const FormInput = ({ handleAction, i18n, children, actionEventTrigger }: FormInputProps) => {
  const [inputValue, setInputValue] = useState<string>('');

  if (!i18n?.label || !handleAction || typeof handleAction !== 'function' || !actionEventTrigger) {
    return null;
  }

  return (
    <form
      className="flex justify-between md:justify-between items-stretch h-10 w-full my-3"
      noValidate
      {...(actionEventTrigger === 'onSubmit' && {
        onSubmit: e => {
          e.preventDefault();
          handleAction(inputValue);
        },
      })}
    >
      <input
        className="mr-4 px-2 min-m-lg border-b-2 rr-border dark:bg-gh-darkly rr-text w-full"
        type="text"
        name="search"
        value={inputValue}
        placeholder={i18n.label}
        onChange={e => {
          setInputValue(e.target.value);
          if (actionEventTrigger === 'onChange') {
            handleAction(e.target.value);
          }
        }}
      />
      {children}
    </form>
  );
};

FormInput.whyDidYouRender = true;
export default FormInput;
