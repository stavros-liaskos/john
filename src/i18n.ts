export const metaI18n = {
  title: 'Release Raccoon',
  description: 'A music release newsletter',
  url: 'https://www.releaseraccoon.online',
};
