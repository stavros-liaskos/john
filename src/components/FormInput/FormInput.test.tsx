import { act, fireEvent, render } from '@testing-library/react';
import FormInput from './FormInput';
import { formInputI18n } from '../../i18n';

describe('FormInput', () => {
  it('renders without data without crashing', () => {
    // @ts-ignore
    render(<FormInput />);
  });

  it('calls action onChange event of input', () => {
    const artistName = 'Cabin Crew';
    const handleAction = jest.fn();
    const { getByRole } = render(
      <FormInput handleAction={handleAction} i18n={formInputI18n} actionEventTrigger={'onChange'} />,
    );
    const inputValue = getByRole('textbox');
    act(() => {
      fireEvent.change(inputValue, { target: { value: artistName } });
    });

    expect(handleAction).toHaveBeenCalledWith(artistName);
  });

  it('calls action onSubmit event of form', () => {
    const artistName = 'Cabin Crew';
    const handleAction = jest.fn();
    const { getByRole } = render(
      <FormInput handleAction={handleAction} i18n={formInputI18n} actionEventTrigger={'onSubmit'}>
        <button>Submit</button>
      </FormInput>,
    );
    const inputValue = getByRole('textbox');
    const btn = getByRole('button');
    act(() => {
      fireEvent.change(inputValue, { target: { value: artistName } });
      fireEvent.click(btn);
    });

    expect(inputValue).toHaveValue(artistName);
    expect(handleAction).toHaveBeenCalledWith(artistName);
  });

  it('matches snapshot', () => {
    const { container } = render(
      <FormInput handleAction={jest.fn} i18n={formInputI18n} actionEventTrigger={'onSubmit'} />,
    );
    expect(container).toMatchSnapshot();
  });
});
