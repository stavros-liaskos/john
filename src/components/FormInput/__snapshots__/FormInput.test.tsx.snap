// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput matches snapshot 1`] = `
<div>
  <form
    class="flex justify-between md:justify-between items-stretch h-10 w-full my-3"
    novalidate=""
  >
    <input
      class="mr-4 px-2 min-m-lg border-b-2 rr-border dark:bg-gh-darkly rr-text w-full"
      name="search"
      placeholder="Here goes the artist.."
      type="text"
      value=""
    />
  </form>
</div>
`;
